### ISSUE TITLE
Indexed Access Types that should be the same as a given types are somehow different than that type, under strictNullChecks

### ISSUE THREAD

**Issue by Ryan1729** [Microsoft/TypeScript#30871]
**TypeScript Version:**  3.5.0-dev.20190411

**Search Terms:**
Indexed Access Types
Lookup Types
strictNullChecks 

**Code**

```ts
type Foo = {bar: undefined} | {bar: string};

const b1: {bar: Foo["bar"]} = {bar: undefined};
const b2: Foo = b1;
const b3: Foo = {bar: undefined};
```

**Expected behavior:**
This should compile with `strictNullChecks` on. I expect `{bar: Foo["bar"]}` to be the same thing as `Foo` in this case.
**Actual behavior:**
running `tsc --strictNullChecks .\bug.ts` result in: 
```
bug.ts:4:7 - error TS2322: Type '{ bar: string | undefined; }' is not assignable
 to type 'Foo'.
  Type '{ bar: string | undefined; }' is not assignable to type '{ bar: string;
}'.
    Types of property 'bar' are incompatible.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

4 const b2: Foo = b1;
        ~~


Found 1 error.
```

**Playground Link:** [Link](https://www.typescriptlang.org/play/index.html#src=type%20Foo%20%3D%20%7Bbar%3A%20undefined%7D%20%7C%20%7Bbar%3A%20string%7D%3B%0D%0A%0D%0Aconst%20b1%3A%20%7Bbar%3A%20Foo%5B%22bar%22%5D%7D%20%3D%20%7Bbar%3A%20undefined%7D%3B%0D%0Aconst%20b2%3A%20Foo%20%3D%20b1%3B%0D%0Aconst%20b3%3A%20Foo%20%3D%20%7Bbar%3A%20undefined%7D%3B)

**Related Issues:** 
https://github.com/Microsoft/TypeScript/issues/29042
https://github.com/Microsoft/TypeScript/issues/21975 This one is extremely similar but it and the ones it was marked as a duplicate of were closed. Is this a regression?

**Comment by jack-williams** [comment]
Might be fixed by #30779

**Comment by RyanCavanaugh** [comment]
> Might be fixed by #30779

Confirmed!
